[Documentaion] updating roctx library linkage documentation (#2251)
ROCm/rocm-systems: projects/rocprofiler-sdk/source/docs/how-to/using-rocprofiler-sdk-roctx.rst

In such cases, it is helpful to define specific tasks to be traced. To specify the tasks for tracing, enclose the respective source code with the API calls provided by the ROCTx library.
This process is also known as instrumentation.

> To use ``rocprofv3`` for marker tracing, including and linking to old ``ROCTx`` works but it's recommended to switch to the new ``ROCTx`` to utilize new APIs.

Reviewer: ```suggestion
To use ``rocprofv3`` for marker tracing, you can include and link to the old ``ROCTx`` ; however, it's recommended to switch to the new ``ROCTx`` to utilize new APIs.
```
